### PR TITLE
Fix notifications for encrypted events

### PR DESCRIPTION
**Pull request by dbkr**
Remember events that we may need to notify for once decrypted and
evaluate them in an Event.decrypted listener.

(Is this the right way to tell if an event is an encrypted event that hasn't yet been decrypted? It seems a little clunky but I could see another way)

### CODE REVIEW

File: src/Notifier.js [hunk 1]
```diff
@@ -244,16 +254,26 @@ const Notifier = {
...
-        if (actions && actions.notify) {
-            if (this.isEnabled()) {
-                this._displayPopupNotification(ev, room);
-            }
-            if (actions.tweaks.sound && this.isAudioEnabled()) {
-                PlatformPeg.get().loudNotification(ev, room);
-                this._playAudioNotification(ev, room);
+        // If it's an encrypted event and the type is still 'm.room.encrypted',
+        // it hasn't yet been decrypted, so wait until it is.
+        if (ev.isEncrypted() && ev.getType() == 'm.room.encrypted') {
```
Comment (richvdh) [inline src/Notifier.js]: You probably want the same as https://github.com/matrix-org/matrix-js-sdk/pull/530/commits/ec5f9a28920a859ec42c990fb1769e5f0860ab56#diff-cf27c1d543e886c89cd9ac8b8aeaf05bR3073.

File: src/Notifier.js [hunk 2]
```diff
@@ -98,8 +105,10 @@ const Notifier = {
         this.boundOnRoomTimeline = this.onRoomTimeline.bind(this);
         this.boundOnSyncStateChange = this.onSyncStateChange.bind(this);
         this.boundOnRoomReceipt = this.onRoomReceipt.bind(this);
+        this.boundOnEventDecrypted = this.onEventDecrypted.bind(this);
         MatrixClientPeg.get().on('Room.timeline', this.boundOnRoomTimeline);
         MatrixClientPeg.get().on('Room.receipt', this.boundOnRoomReceipt);
+        MatrixClientPeg.get().on('Event.decrypted', this.boundOnEventDecrypted);
```
Comment (richvdh) [inline src/Notifier.js]: can we not attach this to the events rather than the whole client?

Comment (dbkr) [inline src/Notifier.js]: We can, although we've mostly been sticking with the pattern of listening for everything via the client (for better or worse), and we'd have to worry about leaking event listeners if they were never decrypted.

File: src/Notifier.js (hunk 1 above)
Comment (dbkr) [inline src/Notifier.js]: ah, looks more sensible. Done.

File: src/Notifier.js (hunk 2 above)
Comment (richvdh) [inline src/Notifier.js]: I'm just not keen on having to iterate the list every time an event is decrypted...

**Pull request merged**